Mathlib/Algebra/GroupWithZero/Basic.lean

```diff
@@ -389,6 +400,21 @@ lemma zero_zpow_eq (n : ℤ) : (0 : G₀) ^ n = if n = 0 then 1 else 0 := by
   · rw [h, zpow_zero]
   · rw [zero_zpow _ h]
 
+lemma eq_zero_of_zero_zpow_eq_one₀ {n : ℤ} : (0 : G₀) ^ n = 1 ↔ n = 0 := by
+  constructor
+  · intro h
+    wlog hn : 0 ≤ n
+    rw [← Int.neg_eq_zero]
+    apply this (G₀ := G₀)
+    · simpa
+    · omega
+    have : n.toNat = 0 := eq_zero_of_zero_pow_eq_one₀ (M₀ := G₀).mp <| by
+      rwa [← zpow_natCast, Int.toNat_of_nonneg hn]
+    omega
+  · intro h
+    rw [h]
+    exact zpow_zero 0
```

Comment:
```suggestion
  rw [zero_zpow_eq, zero_ne_one.ite_eq_left]
```

Reply:
same here, I'm getting an error message if I try the code

Reply:
Oh the lemma is called `Ne.ite_eq_left_iff`, so I guess
```suggestion
  rw [zero_zpow_eq, zero_ne_one.ite_eq_left_iff]
```

Reply:
Well and I guess I need to use `one_ne_zero` and not `zero_ne_one`.